### PR TITLE
Unified Calendar

### PR DESCRIPTION
**Pull request by hiimchrislim**
Created a draft pr to get a feature preview link

### CODE REVIEW

**Comment by hiimchrislim** [comment]
image
Is it possible if we could have the buttons on the bar removed? (i.e. the pencil, heart and triple dot button)

**Comment by milindvishnoi** [comment]
@hiimchrislim I could not assign you for PR review as you started the PR although I have added @yuhengsu to review it.

File: pages/calendar/index.vue [hunk 1]
```diff
@@ -161,35 +158,52 @@ export default {
...
-          end: second,
-          color: this.colors[this.names[club]],
-          timed: !allDay
-        })
+    readCSVData (data, clubName) {
+      console.log(data)
+      // checks if the event data is returned properly
+      if (typeof data === 'string' && data.length > 0) {
+        const lines = data.split('\n')
+        const output = []
```
Comment (ArsalaBangash) [inline pages/calendar/index.vue:166]: What does output denote here?

File: pages/calendar/index.vue [hunk 2]
```diff
@@ -1,109 +1,101 @@
...
+                  <v-spacer></v-spacer>
+                </v-toolbar>
+                <v-card-text>
+                  <span style="font-size: 16px;" v-html="selectedEvent.details"></span>
+                </v-card-text>
+                <v-card-actions>
+                  <v-btn
+                    text
+                    :color="selectedEvent.color"
+                    @click="selectedOpen = false"
```
Comment (ArsalaBangash) [inline pages/calendar/index.vue:87]: Maybe selectedEventFocused is a better var name than selectedOpen?

**Review by ArsalaBangash — CHANGES_REQUESTED** [review]
Let's iterate

**Comment by milindvishnoi** [comment]
@ArsalaBangash Please provide a more comprehensive PR as you mentioned in our last meeting.

File: content/calendar/CSSC.csv [hunk 3]
```diff
@@ -0,0 +1,3 @@
+name, description, start_date, start_time, end_date, end_time
```
Comment (ArsalaBangash) [inline content/calendar/CSSC.csv:1]: I thought we're getting these events from another repo. If these files are duplicated, let's remove them

File: pages/calendar/index.vue [hunk 4]
```diff
@@ -1,109 +1,101 @@
...
+                    color="#ffffff"
+                    v-bind="attrs"
+                    v-on="on"
+                  >
+                    <span>{{ typeToLabel[type] }}</span>
+                    <v-icon right>mdi-menu-down</v-icon>
+                  </v-btn>
+                </template>
+                <v-list>
+                  <v-list-item @click="type = 'day'">
```
Comment (ArsalaBangash) [inline pages/calendar/index.vue:34]: This looks like a fine candidate for a v-for

File: pages/calendar/index.vue [hunk 5]
```diff
@@ -1,109 +1,101 @@
...
+              color="var(--color-primary-light)"
+              :events="monthEvents"
+              :event-color="getEventColor"
+              event-text-color="#ffffff"
+              :type="type"
+              @click:event="showEvent"
+              @click:more="viewDay"
+              @click:date="viewDay"/>
+            <v-menu
+              v-model="selectedOpen"
```
Comment (ArsalaBangash) [inline pages/calendar/index.vue:63]: selectedEventFocused here as well

File: pages/calendar/index.vue [hunk 6]
```diff
@@ -161,35 +158,52 @@ export default {
...
-          name: this.names[club],
-          start: first,
-          end: second,
-          color: this.colors[this.names[club]],
-          timed: !allDay
-        })
+    readCSVData (data, clubName) {
+      console.log(data)
+      // checks if the event data is returned properly
+      if (typeof data === 'string' && data.length > 0) {
```
Comment (ArsalaBangash) [inline pages/calendar/index.vue:164]: Could the data be anything other than string? What are we checking for here?

**Review by ArsalaBangash — CHANGES_REQUESTED** [review]
This is a solid elevation. Let's address the comments from my previous and current review, and we'll proceed further. :)

**Pull request merged**